### PR TITLE
loopdb: fix leapyear parsing

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR fixes https://github.com/lightninglabs/loop/issues/655 by adding a check if a date would be in a leap year.

### CODE REVIEW

**Review by bhandras — APPROVED** [review]
LGTM, thanks for the fix 🥇

File: loopdb/sqlite.go [hunk 1]
```diff
@@ -342,6 +342,26 @@ func fixTimeStamp(dateTimeStr string) (time.Time, error) {
...
+	// If the year is a leap year and the date is 29th of February, we
+	// need to change it to 28th of February. Otherwise, the time.Parse
+	// function will fail, as a non-leap year cannot have 29th of February.
+	day, month, err := extractDayAndMonth(dateTimeStr)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("unable to parse timestamp day "+
+			"and month %v: %v", dateTimeStr, err)
+	}
+
+	if !isLeapYear(thisYear) && isLeapYear(year) &&
```
Comment (hieblmi) [inline loopdb/sqlite.go:354]: couldn't we just replace this if `thisYear` is not a leap year?
`if !isLeapYear(thisYear) && month == 2 && day == 29 {...`

File: loopdb/sqlite.go [hunk 2]
```diff
@@ -403,3 +423,37 @@ func getTimeStampYear(dateTimeStr string) (int, error) {
 
 	return year, nil
 }
+
+func extractDayAndMonth(dateStr string) (int, int, error) {
+	// Split the date string into parts using various delimiters
```
Comment (hieblmi) [inline loopdb/sqlite.go:428]: nit: period

File: loopdb/sqlite.go [hunk 3]
```diff
@@ -403,3 +423,37 @@ func getTimeStampYear(dateTimeStr string) (int, error) {
...
+	parts := strings.FieldsFunc(dateStr, func(r rune) bool {
+		return r == '-' || r == ' ' || r == 'T' || r == ':' || r == '+' || r == 'Z'
+	})
+
+	if len(parts) < 3 {
+		return 0, 0, fmt.Errorf("Invalid date format: %s", dateStr)
+	}
+
+	// Extract year, month, and day from the parts
```
Comment (hieblmi) [inline loopdb/sqlite.go:437]: nit: period

**Review by hieblmi — COMMENTED** [review]
Thanks for the fix! Got one Q.

**Review by hieblmi — APPROVED** [review]
LGTM 🚀

**Pull request merged**